spec/models/concerns/user/low_balance_fraud_check_spec.rb

```diff
@@ -108,4 +108,173 @@
+              @creator.check_for_balance_recovery_and_mark_compliant
+
+              expect(@creator.reload.compliant?).to eq(true)
+              expect(@creator.comments.last.content).to include("Automatically restored to compliant state")
+              expect(@creator.comments.last.content).to include("balance recovered above $100")
+              expect(@creator.comments.last.author_name).to eq("LowBalanceFraudCheck")
+            end
+
+            it "enables refunds for the creator" do
+              @creator.check_for_balance_recovery_and_mark_compliant
+
+              expect(@creator.reload.refunds_disabled?).to eq(false)
+            end
+          end
+        end
+
+        context "when the user was previously not_reviewed" do
+          before do
+            @creator.update!(user_risk_state: "not_reviewed")
+            @creator.send(:disable_refunds_and_put_on_probation!)
+          end
+
+          context "when the unpaid balance is above $100" do
+            before do
+              allow(@creator).to receive(:unpaid_balance_cents).and_return(150_00)
+            end
+
+            it "restores the creator to not_reviewed state" do
+              @creator.check_for_balance_recovery_and_mark_compliant
+
+              expect(@creator.reload.not_reviewed?).to eq(true)
+              expect(@creator.comments.last.content).to include("Automatically restored to not_reviewed state")
+              expect(@creator.comments.last.content).to include("balance recovered above $100")
+              expect(@creator.comments.last.author_name).to eq("LowBalanceFraudCheck")
+            end
+
+            it "enables refunds for the creator" do
+              @creator.check_for_balance_recovery_and_mark_compliant
+
+              expect(@creator.reload.refunds_disabled?).to eq(false)
+            end
+          end
+        end
+
+        context "when the unpaid balance is exactly $100" do
+          before do
+            @creator.update!(user_risk_state: "compliant")
+            @creator.send(:disable_refunds_and_put_on_probation!)
+            allow(@creator).to receive(:unpaid_balance_cents).and_return(100_00)
+          end
+
+          it "does not restore the creator" do
+            @creator.check_for_balance_recovery_and_mark_compliant
+
+            expect(@creator.reload.on_probation?).to eq(true)
+          end
+        end
+
+        context "when the unpaid balance is below $100" do
+          before do
+            @creator.update!(user_risk_state: "compliant")
+            @creator.send(:disable_refunds_and_put_on_probation!)
+            allow(@creator).to receive(:unpaid_balance_cents).and_return(50_00)
+          end
+
+          it "does not restore the creator" do
+            @creator.check_for_balance_recovery_and_mark_compliant
+
+            expect(@creator.reload.on_probation?).to eq(true)
+          end
+        end
+
+        context "when previous state is missing from json_data" do
```

Comment:
This test is outdated, we should now cover the previous states presents in the user's versions